### PR TITLE
Enable Jenkins Security Scan

### PR DESCRIPTION
**Pull request by strangelookingnerd**
It is stated in the `JENKINS SECURITY POLICY` that

> The Jenkins project takes security seriously. We make every possible effort to ensure users can adequately secure their automation infrastructure. To that end, we work with Jenkins core and plugin developers, as well as security researchers, to fix security vulnerabilities in Jenkins in a timely manner, and to improve the security of Jenkins in general.

We as plugin developers should try our best to support this. One way of doing so is to enable the Jenkins Security Scan on this repository. For more details see the [documentation](https://www.jenkins.io/doc/developer/security/scan/).

This PR may create new issues in the Code scanning section of this project once it is merged. I'd be happy to try and help resolve those if necessary.

_Please note this PR was created semi-automatically. If you find any issue with it, don't hesitate to ping me._
_For more details see the [Jenkins Developers Google Group](https://groups.google.com/u/1/g/jenkinsci-dev/c/cxjQWnZ0S58)._

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by aytey** [comment]
@strangelookingnerd

> This PR may create new issues in the Code scanning section of this project once it is merged. I'd be happy to try and help resolve those if necessary.
>

Why would that be?

File: .github/workflows/jenkins-security-scan.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Jenkins Security Scan
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+  workflow_dispatch:
```
Comment (aytey) [inline .github/workflows/jenkins-security-scan.yml:9]: This section looks good to me, and has everything I'd expect.

File: .github/workflows/jenkins-security-scan.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  security-events: write
+  contents: read
+  actions: read
+
+jobs:
+  security-scan:
+    uses: jenkins-infra/jenkins-security-scan/.github/workflows/jenkins-security-scan.yaml@v2
+    with:
+      java-cache: 'maven' # Optionally enable use of a build dependency cache. Specify 'maven' or 'gradle' as appropriate.
+      # java-version: 21 # Optionally specify what version of Java to set up for the build, or remove to use a recent default.
```
Comment (aytey) [inline .github/workflows/jenkins-security-scan.yml:21]: I am not sure why we have this commented out -- let's nuke it.

**Comment by strangelookingnerd** [comment]
> Why would that be?

This is how the scan behaves, as it is using CodeQL which is utilizing the Code scanning section.

**Comment by aytey** [comment]
> > Why would that be?
> 
> This is how the scan behaves, as it is using CodeQL which is utilizing the Code scanning section.

Sorry, this doesn't really help.

Are you saying CodeQL will only report findings against `master` and not against a feature branch? If so, how do we configure it to report against a feature branch _before that branch lands_?

**Comment by strangelookingnerd** [comment]
> Are you saying CodeQL will only report findings against `master` and not against a feature branch? If so, how do we configure it to report against a feature branch _before that branch lands_?

No, it will report issues for all commit to the [master branch](https://github.com/jenkinsci/vectorcast-execution-plugin/pull/52/files/2c8710fd50cc44a53cbda166bd682f46618eaf05#diff-0b410817e3c20818859b587da6aeb232da770a20f536c60f4a655a95902a0c76R6) as well as all pushes on [pull requests](https://github.com/jenkinsci/vectorcast-execution-plugin/pull/52/files/2c8710fd50cc44a53cbda166bd682f46618eaf05#diff-0b410817e3c20818859b587da6aeb232da770a20f536c60f4a655a95902a0c76R7) unrelated of the target.

File: .github/workflows/jenkins-security-scan.yml (hunk 2 above)
Comment (TimSVector) [reply .github/workflows/jenkins-security-scan.yml]: I’ll take this into consideration for the 0.78 release.

**Comment by aytey** [comment]
> > Are you saying CodeQL will only report findings against `master` and not against a feature branch? If so, how do we configure it to report against a feature branch _before that branch lands_?
> 
> No, it will report issues for all commit to the [master branch](https://github.com/jenkinsci/vectorcast-execution-plugin/pull/52/files/2c8710fd50cc44a53cbda166bd682f46618eaf05#diff-0b410817e3c20818859b587da6aeb232da770a20f536c60f4a655a95902a0c76R6) as well as all pushes on [pull requests](https://github.com/jenkinsci/vectorcast-execution-plugin/pull/52/files/2c8710fd50cc44a53cbda166bd682f46618eaf05#diff-0b410817e3c20818859b587da6aeb232da770a20f536c60f4a655a95902a0c76R7) unrelated of the target.

Okay, so it shouldn't find new things _for `master`_, but it might find new things _for pending PRs_?

**Review by TimSVector — APPROVED** [review]
Adding to 0.78 release 

branch: tms_078

**Pull request closed without merging**